shell/platform/common/cpp/accessibility_bridge.h

```diff
@@ -0,0 +1,266 @@
+/// FlutterPlatformNodeDelegate and AccessibilityBridgeDelegate.
+class AccessibilityBridge : public ui::AXTreeObserver {
+ public:
+  //-----------------------------------------------------------------------------
+  /// Delegate to handle requests from the accessibility bridge. The requests
+  /// include sending accessibility event to native accessibility system,
+  /// routing accessibility action to the Flutter framework, and creating
+  /// platform specific FlutterPlatformNodeDelegate.
+  ///
+  /// The accessibility events are generated when accessibility tree changes.
+  /// These events must be sent to the native accessibility system through
+  /// the native API for the system to pick up the changes
+  /// (e.g. NSAccessibilityPostNotification in MacOS).
+  ///
+  /// The accessibility actions are generated by the native accessibility system
+  /// when users interacted with the assistive technologies. Those actions
+  /// needed to be sent to the Flutter framework.
+  ///
+  /// Each platform needs to implement the FlutterPlatformNodeDelegate and
+  /// returns its platform specific instance of FlutterPlatformNodeDelegate
+  /// in this delegate.
+  class AccessibilityBridgeDelegate {
+   public:
+    virtual ~AccessibilityBridgeDelegate() = default;
+    //---------------------------------------------------------------------------
+    /// @brief      Handle accessibility events generated due to accessibility
+    ///             tree changes. These events are generated in accessibility
+    ///             bridge and needed to be sent to native accessibility system.
+    ///             See ui::AXEventGenerator::Event for possible events.
+    ///
+    /// @param[in]  targeted_event      The object that contains both the
+    ///                                 generated event and the event target.
+    virtual void OnAccessibilityEvent(
+        ui::AXEventGenerator::TargetedEvent targeted_event) = 0;
+
+    //---------------------------------------------------------------------------
+    /// @brief      Dispatch accessibility action back to the Flutter framework.
+    ///             These actions are generated in the native accessibility
+    ///             system when users interact with the assistive technologies.
+    ///             For example, a
+    ///             FlutterSemanticsAction::kFlutterSemanticsActionTap is
+    ///             fired when user click or touch the screen.
+    ///
+    /// @param[in]  target              The semantics node id of the action
+    ///                                 target.
+    /// @param[in]  action              The generated flutter semantics action.
+    /// @param[in]  data                Additional data associated with the
+    ///                                 action.
+    /// @param[in]  data_size           The length of the additional data.
+    virtual void DispatchAccessibilityAction(ui::AXNode::AXID target,
```

Comment:
nit (and elsewhere): would it be more ergonomic for developers to have a `flutter::` namespaced typedef for this?

`AXID` and `AXNode` seem like things we don't want to be exposing directly to developers.